packages/core/src/__tests__/ceramic-anchor.test.ts

```diff
@@ -314,30 +318,47 @@ describe('Ceramic anchoring', () => {
     await ceramic2.close()
   })
 
-  it('test the same state anchored twice, first one wins', async () => {
-    await ipfs2.swarm.connect(multaddr1)
+  it('test the same state anchored twice, last one wins (in-memory anchor service behavior)', async () => {
+    await ipfs3.swarm.connect(multaddr1)
+    await ipfs3.swarm.connect(multaddr2)
 
-    const [ceramic1, ceramic2] = await Promise.all([
+    const [ceramic1, ceramic2, ceramic3] = await Promise.all([
       createCeramic(ipfs1, true),
-      createCeramic(ipfs2, false)
+      createCeramic(ipfs2, true),
+      createCeramic(ipfs3, true),
     ])
     const controller = ceramic1.context.did.id
 
-    const doctype1 = await ceramic1.createDocument(DOCTYPE_TILE, { content: { x: 1 } }, { anchor: false, publish: false })
-    const cloned = new TileDoctype(doctype1.state, doctype1.context)
-    await doctype1.change({ content: { x: 7 }, metadata: { controllers: [controller] } }, { anchor: true, publish: true })
-    await cloned.change({ content: { x: 5 }, metadata: { controllers: [controller] } }, { anchor: true, publish: true })
+    ceramic1.context.anchorService = ceramic3.context.anchorService // use ceramic3 in-memory anchor service
+    ceramic2.context.anchorService = ceramic3.context.anchorService // use ceramic3 in-memory anchor service
 
-    await anchorDoc(ceramic1, doctype1)
+    const doctype1 = await ceramic1.createDocument(DOCTYPE_TILE, { content: { x: 1 } }, { anchor: false, publish: true })
+    const doctype2 = await ceramic2.loadDocument(doctype1.id)
+    await doctype1.change({ content: { x: 7 }, metadata: { controllers: [controller] } }, { anchor: true, publish: false })
+    await doctype2.change({ content: { x: 5 }, metadata: { controllers: [controller] } }, { anchor: true, publish: false })
+
+    const handle1 = registerChangeListener(doctype1)
+    const handle2 = registerChangeListener(doctype2)
 
-    expect(doctype1.content).toEqual({ x: 7 })
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    await ceramic3.context.anchorService.anchor()
+
+    await handle1
+    await handle2
+
+    expect(doctype1.content).toEqual({ x: 5 })
     expect(doctype1.state.log.length).toEqual(3)
 
-    const doctype2 = await ceramic2.loadDocument(doctype1.id)
-    expect(doctype1.content).toEqual(doctype2.content)
-    expect(doctype1.state.log.length).toEqual(doctype2.state.log.length)
+    expect(doctype2.content).toEqual({ x: 5 })
```

Comment:
This is specifically relying on the behavior of the in-memory anchor service to pick which logs to anchor based on timestamp rather than based on deterministic conflict resolution strategies like we'll actually use in real anchor services. If we're going to write a test of conflict resolution behavior, it should be based on the behaviors we actually want and expect